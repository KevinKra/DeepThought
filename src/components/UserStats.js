import React from "react";
import ReviewButton from "./ReviewButton";
import "../styles/_UserStats.scss";
class UserStats extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      totalReviewed: this.props.statsTotalReviewed
    };
  }
  render() {
    return (
      <section className="user-stats">
        <article>Total cards Reviewed:{this.state.totalReviewed}</article>
        <ReviewButton purpose="Review Cards" />
        <ReviewButton purpose="Starred Cards" />
      </section>
    );
  }
}

export default UserStats;
