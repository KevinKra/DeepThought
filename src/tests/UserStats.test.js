import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import UserStats from "../components/UserStats";

describe("<UserStats />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<UserStats />);
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
