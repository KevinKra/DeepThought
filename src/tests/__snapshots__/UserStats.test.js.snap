// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStats /> should match the snapshot 1`] = `
<section
  className="user-stats"
>
  <article>
    Total cards Reviewed:
  </article>
  <ReviewButton
    purpose="Review Cards"
  />
  <ReviewButton
    purpose="Starred Cards"
  />
</section>
`;
